integration/src/test/java/edu/unc/lib/boxc/integration/web/access/CollectionsEndpointIT.java

```
@@ -72,35 +77,121 @@ public class CollectionsEndpointIT {
+
+        try (var resp = httpClient.execute(getMethod)) {
+            var metadata = getMetadataFromResponse(resp);
+
+            assertSuccessfulResponse(resp);
+            assertValuePresent(metadata, 0, "type", "AdminUnit");
+            assertValuePresent(metadata, 1, "type", "AdminUnit");
+        }
+    }
+
+    public void testCollectionsJsonReturnsCorrectTitle() throws Exception {
+        adminUnitFactory.createAdminUnit(Map.of("title", "Object2"));
+
+        try (var resp = httpClient.execute(getMethod)) {
+            var metadata = getMetadataFromResponse(resp);
+
+            assertSuccessfulResponse(resp);
+            assertValuePresent(metadata, 0, "title", "Object2");
+        }
+    }
+
+    public void testCollectionsJsonReturnsThumbnailUrl() throws Exception {
+        var options = Map.of("title", "Object1", "addThumbnail", "true");
+        adminUnitFactory.createAdminUnit(options);
+
+        try (var resp = httpClient.execute(getMethod)) {
+            var metadata = getMetadataFromResponse(resp);
+
+            assertSuccessfulResponse(resp);
+            assertValuePresent(metadata, 0, "thumbnail_url");
+        }
+    }
+
+    public void testCollectionsJsonReturnsChildrenCount() throws Exception {
+        var options = Map.of("title", "Object1", "addThumbnail", "true");
         var adminUnit = adminUnitFactory.createAdminUnit(options);
-        var collection = collectionFactory.createCollection(adminUnit, options);
-        var work = workFactory.createWork(collection, options);
+        collectionFactory.createCollection(adminUnit,
+                Map.of("title", "Collection1", "readGroup", PUBLIC_PRINC));
+
+        try (var resp = httpClient.execute(getMethod)) {
+            var metadata = getMetadataFromResponse(resp);
+            var childCount = metadata.get(0).get("counts").get("child").asInt();
+
+            assertSuccessfulResponse(resp);
+            assertValuePresent(metadata, 0, "counts");
```

this is probably redundant since line 150 will fail if "counts" isn't populated

---

so we don't need an assertion of it will fail if the condition isn't met? that seems very hidden

---

You could move the assignment of childCount to after the value check, so we would get a assertion failure rather than a NullPointerException

---

i'll just delete it